src/backend/distributed/commands/multi_copy.c

```diff
@@ -2468,6 +2468,10 @@ CitusSendTupleToPlacements(TupleTableSlot *slot, CitusCopyDestReceiver *copyDest
 
 		if (switchToCurrentPlacement)
 		{
+			List *placementList = ShardPlacementList(shardId);
```

Comment:
This is necessary because otherwise COPY code-path wouldn't call `AssociatePlacementAccessWithRelation ()` at all. I think this is a decent place to have this call.

Reply:
Wouldn't this happen quite often?

Reply:
For COPY without in tx block, once per placement. Because, we'll be only switching to the current placement once at the beginning of the COPY command.

For COPY (or anything that copies into a distributed table) inside a tx, if the previous command multiplexed some commands over a single connection, then we'll get to here once in 4MB COPY into the placement as well.

I guess this is okayish, because we don't have many choices here since we don't know which placement are going to be accessed by the COPY. 

Especially with #2867, this'll be almost a no-op so I didn't want to add any extra logic to keep track of which placements are already accessed.

Reply:
`ShardPlacementList ()` is not a free function even if `AssociatePlacementAccessWithRelation ` could be. One thing I considered was to have another api for `AssociatePlacementAccessWithRelation ()`, which gets the shard itself. In fact, we're only using the `shardId` there. 

Do you have any preference here?